src/main/java/de/rwth/idsg/steve/repository/impl/TransactionRepositoryImpl.java

```diff
@@ -209,162 +268,76 @@ public TransactionDetails getDetails(int transactionPk) {
-                CONNECTOR.CHARGE_BOX_ID,
-                CONNECTOR.CONNECTOR_ID,
-                TRANSACTION.ID_TAG,
-                TRANSACTION.START_TIMESTAMP,
-                TRANSACTION.START_VALUE,
-                TRANSACTION.STOP_TIMESTAMP,
-                TRANSACTION.STOP_VALUE,
-                TRANSACTION.STOP_REASON
-        );
-
-        return addConditions(selectQuery, form);
-    }
+    private List<Condition> getConditions(TransactionQueryForm form) {
+        List<Condition> conditions = new ArrayList<>();
 
-    /**
-     * Difference from getInternalCSV:
-     * Joins with CHARGE_BOX and OCPP_TAG tables, selects CHARGE_BOX_PK and OCPP_TAG_PK additionally
-     */
-    @SuppressWarnings("unchecked")
-    private
-    SelectQuery<Record12<Integer, String, Integer, String, DateTime, String, DateTime, String, String, Integer, Integer, TransactionStopEventActor>>
-    getInternal(TransactionQueryForm form) {
-
-        SelectQuery selectQuery = ctx.selectQuery();
-        selectQuery.addFrom(TRANSACTION);
-        selectQuery.addJoin(CONNECTOR, TRANSACTION.CONNECTOR_PK.eq(CONNECTOR.CONNECTOR_PK));
-        selectQuery.addJoin(CHARGE_BOX, CHARGE_BOX.CHARGE_BOX_ID.eq(CONNECTOR.CHARGE_BOX_ID));
-        selectQuery.addJoin(OCPP_TAG, OCPP_TAG.ID_TAG.eq(TRANSACTION.ID_TAG));
-        selectQuery.addSelect(
-                TRANSACTION.TRANSACTION_PK,
-                CONNECTOR.CHARGE_BOX_ID,
-                CONNECTOR.CONNECTOR_ID,
-                TRANSACTION.ID_TAG,
-                TRANSACTION.START_TIMESTAMP,
-                TRANSACTION.START_VALUE,
-                TRANSACTION.STOP_TIMESTAMP,
-                TRANSACTION.STOP_VALUE,
-                TRANSACTION.STOP_REASON,
-                CHARGE_BOX.CHARGE_BOX_PK,
-                OCPP_TAG.OCPP_TAG_PK,
-                TRANSACTION.STOP_EVENT_ACTOR
-        );
-
-        return addConditions(selectQuery, form);
-    }
-
-    @SuppressWarnings("unchecked")
-    private SelectQuery addConditions(SelectQuery selectQuery, TransactionQueryForm form) {
         if (form.isTransactionPkSet()) {
-            selectQuery.addConditions(TRANSACTION.TRANSACTION_PK.eq(form.getTransactionPk()));
+            conditions.add(TRANSACTION.TRANSACTION_PK.eq(form.getTransactionPk()));
         }
 
         if (form.isChargeBoxIdSet()) {
-            selectQuery.addConditions(CONNECTOR.CHARGE_BOX_ID.eq(form.getChargeBoxId()));
+            conditions.add(CONNECTOR.CHARGE_BOX_ID.eq(form.getChargeBoxId()));
         }
 
         if (form.isOcppIdTagSet()) {
-            selectQuery.addConditions(TRANSACTION.ID_TAG.eq(form.getOcppIdTag()));
+            conditions.add(TRANSACTION.ID_TAG.eq(form.getOcppIdTag()));
         }
 
         if (form.isUserIdSet()) {
-            var query = ocppTagByUserIdQuery(ctx, form.getUserId());
-            selectQuery.addConditions(TRANSACTION.ID_TAG.in(query));
+            conditions.add(USER_OCPP_TAG.USER_PK.eq(form.getUserId()));
```

Comment:
since we are left-joining with the `USER_OCPP_TAG` table, we have direct access to `userId` in the original query now. `ocppTagByUserIdQuery(ctx, form.getUserId());` is not needed anymore.